python/pyarrow/tests/test_fs.py

```diff
@@ -443,6 +445,12 @@ def test_s3fs_limited_permissions_create_bucket(s3_server):
     )
     fs.create_dir('existing-bucket/test')
 
+    with pytest.raises(pa.ArrowIOError):
```

Comment:
Perhaps also add a `match=` argument to also check relevant part of the error message?